js/symbol/shaping.js

```diff
@@ -186,11 +186,13 @@ function align(positionedGlyphs, justify, horizontalAlign, verticalAlign, maxLin
     }
 }
 
-function shapeIcon(image, layout) {
+function shapeIcon(image, layout, layer, featureProperties) {
```

Comment:
As of this diff, the `layout` argument is entirely unused by `shapeIcon`. Could you please remove it?

It may be cleaner to pass `iconOffset` to this function directly instead of `layer` and `featureProperties`. (["stamp coupling"](https://en.wikipedia.org/wiki/Coupling_(computer_programming)))